docs/admin/kubeadm-init.md

```diff
@@ -0,0 +1,425 @@
+  image: <string>
+networking:
+  dnsDomain: <string>
+  serviceSubnet: <cidr>
+  podSubnet: <cidr>
+kubernetesVersion: <string>
+cloudProvider: <string>
+nodeName: <string>
+authorizationModes:
+- <authorizationMode1|string>
+- <authorizationMode2|string>
+token: <string>
+tokenTTL: <time duration>
+selfHosted: <bool>
+apiServerExtraArgs:
+  <argument>: <value|string>
+  <argument>: <value|string>
+controllerManagerExtraArgs:
+  <argument>: <value|string>
+  <argument>: <value|string>
+schedulerExtraArgs:
+  <argument>: <value|string>
+  <argument>: <value|string>
+apiServerCertSANs:
+- <name1|string>
+- <name2|string>
+certificatesDir: <string>
+imageRepository: <string>
+unifiedControlPlaneImage: <string>
+featureGates:
+  <feature>: <bool>
+  <feature>: <bool>
+```
+
+### Passing custom arguments to control plane components {#custom-args}
+
+If you would like to override or extend the behaviour of a control plane component, you can provide 
+extra arguments to kubeadm. When the component is deployed, it will use these additional arguments in 
+the pod command itself. 
+
+For example, to add additional feature-gate arguments to the apiserver, your [configuration file](#sample-master-configuration) 
+will need to look like this:
+
+```
+apiVersion: kubeadm.k8s.io/v1alpha1
+kind: MasterConfiguration
+apiServerExtraArgs:
+   feature-gates: APIResponseCompression=true
+```
+
+To customise the scheduler or controller-manager, use `schedulerExtraArgs` and `controllerManagerExtraArgs` respectively.
+
+More information on custom arguments can be found here:
+- [kube-apiserver](https://kubernetes.io/docs/admin/kube-apiserver/)
+- [kube-controller-manager](https://kubernetes.io/docs/admin/kube-controller-manager/)
+- [kube-scheduler](https://kubernetes.io/docs/admin/kube-scheduler/)
+
+### Using custom images {#custom-images}
+
+By default, kubeadm will pull images from `gcr.io/google_containers`, unless
+the requested Kubernetes version is a CI version. In this case,
+`gcr.io/kubernetes-ci-image` will be used.
+
+This behaviour can be overridden by [using kubeadm with a configuration file](#config-file).
+Allowed customization are:
+
+* provide an alternative `imageRepository` to be used instead of
+  `gcr.io/google_containers`
+* provide an `unifiedControlPlaneImage` to be used instead of different images for each control plane component
+  for control plane components
+* provide a specific `etcd.image` to be used instead of the image available at`gcr.io/google_containers`
+
+
+### Using custom certificates {#custom-certificates}
+
+By default kubeadm will generate all the certificates needed for a cluster to run.
+You can override this behaviour by providing your own certificates.
+
+To do so, you must place them in whatever directory is specified by the
+`--cert-dir` flag or `CertificatesDir` configuration file key. By default this
+is `/etc/kubernetes/pki`.
+
+If a given certificate and private key pair both exist, kubeadm will skip the
```

Comment:
Either say, "If a given certificate and private key both exist", or say "If a given certificate and private key pair exists".